Framework/DataHandling/src/LoadSampleShape.cpp

```diff
@@ -0,0 +1,222 @@
+#include "MantidDataHandling/LoadSampleShape.h"
+
+#include "MantidAPI/AnalysisDataService.h"
+#include "MantidAPI/FileProperty.h"
+#include "MantidAPI/InstrumentValidator.h"
+#include "MantidAPI/WorkspaceFactory.h"
+#include "MantidAPI/MatrixWorkspace.h"
+#include "MantidAPI/MatrixWorkspace_fwd.h"
+#include "MantidAPI/Sample.h"
+
+#include "MantidGeometry/Instrument.h"
+#include "MantidKernel/Material.h"
+#include "MantidKernel/CompositeValidator.h"
+#include "MantidKernel/EnabledWhenProperty.h"
+#include "MantidKernel/Exception.h"
+
+#include <Poco/File.h>
+
+namespace Mantid {
+namespace DataHandling {
+
+// Register the algorithm into the algorithm factory
+DECLARE_ALGORITHM(LoadSampleShape)
+
+using namespace Kernel;
+using namespace API;
+using namespace Geometry;
+
+namespace {
+
+bool areEqualVertices(Kernel::V3D const &v1, Kernel::V3D const &v2) {
+  Kernel::V3D diff = v1 - v2;
+  return diff.norm() < 1e-9;
```

Comment:
You should include a comment here explaining what 2e-9 means in real world measurements, given that the base unit of distance in Mantid is m.